flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogStore.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.catalog.exceptions.CatalogException;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * The {@link CatalogStore} is used in {@link CatalogManager} to retrieve, save and remove {@link
+ * CatalogDescriptor} at the external storage system.
+ */
```

Comment:
let's try to define the interface firstly instead of explain it.
```suggestion
/**
 * Represents the storage where persists all {@link Catalog}s.
 *
 * <p>All catalogs can be lazy initialized with the {@link CatalogStore}.
 *
 * <p>It can be used in {@code CatalogManager} to retrieve, save and remove catalog in {@link
 * CatalogDescriptor} format at the external storage system.
 */
```